astropy/modeling/fitting.py

```diff
@@ -1068,3 +1070,35 @@ def _validate_model(model, supported_constraints):
 
     model_copy = model.copy()
     return model_copy
+
+
+def populate_entry_points(entry_points):
```

Comment:
@pllim - we had some discussion about whether to turn this entry points functionality into a generalized utility as you suggest.  In the end we decided that since there is currently only one use case that it would be sensible to leave this as a dedicated function for fitting.  Then when/if there are new use cases for this in other sub-packages it will be easier to make decisions about API.